catgrad-core/examples/hidden.rs

```diff
@@ -80,6 +97,51 @@ fn run_interpreter(
     Ok(())
 }
 
+#[cfg(feature = "candle-backend")]
+fn run_interpreter_candle(
```

Comment:
Instead of duplicating the `run_interpreter` code, please make `run_interpreter` generic instead, e.g.:

```rust
fn run_interpreter<B: Backend>(
    backend: B,
    typed_term: &TypedTerm,
    env: Environment,
) -> Result<(), Box<dyn std::error::Error>> {
    todo!()
}
```